### DISCUSSION TITLE
icon name match language name from GitHub API

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [devicons/devicon#2266] [Ideas]
Would it be possible for your language names to match the GitHub API names? Looking at the json file it looks like you support alternative names, so maybe it would be kind of easy to add? All the languages are listed here: https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml (as the primary keys)

Basically the idea would be that you can grab info from the github api, and without doing any mapping, and automatically know the language icon.

A few years ago I did this based on the same list/names: https://github.com/LizardByte/uno/tree/master/gh-pages-template/language-icons ... I don't really want to keep this though, and would rather use a maintained project like devicon.

**Comment by Snailedlt** [comment]
This would be a good feature imo, though I'm not sure how much work would be required to make it happen. I'm not going to work on this myself since there are many tasks with higher priority, but feel free to work on this and make a PR if you'd like :)

**Reply by ReenigneArcher** [reply]
I have a draft PR here: https://github.com/devicons/devicon/pull/2270

I have ~170 more icons to add.
